azext_iot/central/commands_device.py

```diff
@@ -9,18 +9,39 @@
 
 from azext_iot.common import utility
 from azext_iot.constants import CENTRAL_ENDPOINT
-from azext_iot.central.providers import CentralDeviceProvider
+from azext_iot.central.providers.preview import CentralDeviceProviderPreview
+from azext_iot.central.providers.v1 import CentralDeviceProviderV1
+from azext_iot.central.models.enum import ApiVersion
 
 
-def list_devices(cmd, app_id: str, token=None, central_dns_suffix=CENTRAL_ENDPOINT):
-    provider = CentralDeviceProvider(cmd=cmd, app_id=app_id, token=token)
+def list_devices(
+    cmd,
+    app_id: str,
+    token=None,
+    central_dns_suffix=CENTRAL_ENDPOINT,
+    api_version=ApiVersion.v1.value,
+):
+    if api_version == ApiVersion.preview.value:
```

Comment:
Not strictly necessary for this PR, but I could see some higher level component, like a central provider manager of sorts, which takes shared params like...cmd, app_id, token, provider type and  api version and returns the proper configuration of the desired provider. The advantage is common logic for choosing the right provider is contained in this component.

Reply:
great suggestion will be useful as we add more versions.